drivers/i2c/i2c_ll_stm32.c

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright (c) 2016 BayLibre, SAS
+ * Copyright (c) 2017 Linaro Ltd
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <clock_control/stm32_clock_control.h>
+#include <clock_control.h>
+#include <misc/util.h>
+#include <kernel.h>
+#include <board.h>
+#include <errno.h>
+#include <i2c.h>
+#include "i2c_ll_stm32.h"
+
+#define SYS_LOG_LEVEL CONFIG_SYS_LOG_I2C_LEVEL
+#include <logging/sys_log.h>
+
+static int i2c_stm32_runtime_configure(struct device *dev, u32_t config)
+{
+	const struct i2c_stm32_config *cfg = DEV_CFG(dev);
+	struct i2c_stm32_data *data = DEV_DATA(dev);
+	I2C_TypeDef *i2c = cfg->i2c;
+	u32_t clock;
+
+	if (data->dev_config.bits.is_slave_read)
+		return -EINVAL;
+
+	data->dev_config.raw = config;
+
+	clock_control_get_rate(device_get_binding(STM32_CLOCK_CONTROL_NAME),
+			(clock_control_subsys_t *) &cfg->pclken, &clock);
+
+	LL_I2C_Disable(i2c);
+	LL_I2C_SetMode(i2c, LL_I2C_MODE_I2C);
+
+	return i2c_configure_timing(dev, clock);
+}
+
+#define OPERATION(msg) (((struct i2c_msg *) msg)->flags & I2C_MSG_RW_MASK)
+
+static int i2c_stm32_transfer(struct device *dev, struct i2c_msg *message,
+				u8_t messages, u16_t slave)
+{
+	const struct i2c_stm32_config *cfg = DEV_CFG(dev);
+	struct i2c_msg *current, *next;
+	I2C_TypeDef *i2c = cfg->i2c;
+	int ret = 0;
+
+	LL_I2C_Enable(i2c);
+
+	current = message;
+	while (messages > 0) {
```

Comment:
I think I for loop is easier to understand:
for (int i = 0; i < num_msgs; i++, msg++)
{

...
}